src/biopsykit/io/psg.py

```diff
@@ -0,0 +1,270 @@
+"""Module for importing data recorded by a PSG system (expects .edf files)."""
+
+import datetime
+import warnings
+from pathlib import Path
+from typing import Dict, Optional, Sequence, Union
+
+import mne
+import pandas as pd
+
+from biopsykit.utils._datatype_validation_helper import _assert_file_extension, _assert_is_dir
+from biopsykit.utils._types import path_t
+
+__all__ = ["PSGDataset"]
+
+
+class PSGDataset:
+    """Class for loading and processing PSG data."""
+
+    def __init__(
+        self,
+        data_dict: Dict[str, pd.DataFrame],
+        sampling_rate_dict: Dict[str, int],
+        start_time: Optional[pd.Timestamp] = None,
+        tz: Optional[str] = None,
+    ):
+
+        self._data = data_dict
+        for name, data in data_dict.items():
+            setattr(self, name, data)
+        for name, sampling_rate in sampling_rate_dict.items():
+            setattr(self, "sampling_rate", sampling_rate)
+        setattr(self, "channels", list(self._data.keys()))
+        self._sampling_rate = sampling_rate_dict
+        self._start_time_unix = start_time
+        self._tz = tz
+
+    @classmethod
+    def from_edf_file(
+        cls,
+        path: path_t,
+        datastreams: Optional[Sequence] = None,
+        tz: Optional[str] = "Europe/Berlin",
+    ):
+        """Create a new Dataset from a valid .edf file.
+
+        Parameters
+        ----------
+        path : :class:`pathlib.Path` or str
+            Path to the .edf file
+        datastreams : Optional[Sequence], optional
+            List of datastreams to load, by default None. If None, all datastreams are loaded.
+        tz : Optional[str], optional
+            Timezone of the recording, by default "Europe/Berlin".
+
+        Returns
+        -------
+        PSGDataset : New Dataset instance
+
+        """
+        if path.is_dir():
+            data_dict, fs, start_time = cls.load_data_folder(path, datastreams, tz)
+        else:
+            data_dict, fs, start_time = cls.load_data(path, datastreams, tz)
+
+        return cls(data_dict=data_dict, sampling_rate_dict={"sampling_rate": fs}, start_time=start_time, tz=tz)
+
+    @property
+    def start_time_unix(self) -> Optional[pd.Timestamp]:
+        """Start time of the recording in UTC time."""
+        return self._start_time_unix
+
+    @property
+    def timezone(self) -> str:
+        """Timezone the dataset was recorded in."""
+        return self._tz
+
+    def data_as_df(self):
```

Comment:
check that index names are consistent with other importers, e.g.:

"samples": plain sample count
"t": relative time in seconds since beginning of recording
"date": absolute time in UTC time
"date (<tz>)": absolute time in localized timezone


see: https://github.com/mad-lab-fau/BioPsyKit/blob/8d346333101a15e2f11a604bff4cc239b7b510eb/src/biopsykit/io/biopac.py#L198